src/Bicep.Core/Semantics/Namespaces/NamespaceResolver.cs

```diff
@@ -61,12 +60,17 @@ void TryAddBuiltInNamespace(string @namespace)
             }
 
             TryAddBuiltInNamespace(SystemNamespaceType.BuiltInName);
-            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile)
+
+            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile) // don't register "az" namespace for Bicep Parameters files
             {
-                // don't register "az" namespace for Bicep Parameters files
-                TryAddBuiltInNamespace(AzNamespaceType.BuiltInName);
+                var foundAzProviderDeclaration = sourceFile.ProgramSyntax.Children
+                    .OfType<ImportDeclarationSyntax>()
+                    .Any(x => x.Alias is not null && x.Alias.NameEquals(AzNamespaceType.BuiltInName));
```

Comment:
Aliases are optional for import declarations. Should we check `x.Specification.Name` instead?